lib/SetAPI/sampleset/SampleSetInterface.py

```diff
@@ -1,65 +1,72 @@
+        """Save a sample set to the workspace.
+
+        :param self: this class
+        :type self: SampleSetInterface
+        :param ctx: KBase context
+        :type ctx: dict[str, Any]
+        :param params: request parameters
+        :type params: dict[str, Any]
+        :return: result of saving sampleset to the workspace
+        :rtype: dict[str, Any]
+        """
+        output_object = deepcopy(params)
+
+        # N.b. according to the spec, these are valid input params.
+        # Just hope no one is trying to use them, eh?
+        output_object["sample_ids"] = []
+        output_object["condition"] = []
         for item in params.get("sample_n_conditions", []):
             item_condition = item["condition"]
             if not isinstance(item_condition, (str | list)):
                 err_msg = "ERROR: condition should be either a list or a string"
-                raise ValueError(err_msg)
+                raise TypeError(err_msg)
 
             if isinstance(
                 item_condition, list
             ):  # Auto populate UI puts input into an array
-                if len(item_condition) > 1:
-                    err_msg = "ERROR: please only select 1 condition per reads object"
+                if len(item_condition) != 1:
+                    err_msg = "ERROR: please select 1 condition per reads object"
                     raise ValueError(err_msg)
                 item["condition"] = item_condition[0]
 
-            params["sample_ids"].extend(item["sample_id"])
+            output_object["sample_ids"].extend(item["sample_id"])
+
+            output_object["condition"].extend(
+                [item["condition"] for _ in item["sample_id"]]
+            )
 
-            params["condition"].extend([item["condition"] for i in item["sample_id"]])
-        pprint(params)
-        ### Create the working dir for the method; change it to a function call
-        out_obj = {k: v for k, v in params.items() if k not in ("ws_id",)}
+        sample_ids = output_object["sample_ids"]
+        output_object["num_samples"] = len(sample_ids)
 
-        sample_ids = params["sample_ids"]
-        out_obj["num_samples"] = len(sample_ids)
-        ## Validation to check if the Set contains more than one samples
+        # check that the sampleset contains at least two samples
         if len(sample_ids) < 2:
             err_msg = "This method takes two (2) or more RNASeq Samples. \
                 If you have only one read sample, run either 'Align Reads using Tophat' or 'Align Reads using Bowtie2' directly for getting alignment"
 
             raise ValueError(err_msg)
 
-        ## Validation to Check if the number of samples is equal to number of condition
-        if len(params["condition"]) != out_obj["num_samples"]:
-            err_msg = "Please specify a treatment label for each sample in the RNA-seq SampleSet. Please enter the same label for the replicates in a sample type"
-            raise ValueError(err_msg)
-
```

Comment:
this is already covered by previous checks so there is no way to trigger this error